### ISSUE TITLE
Feature: Allow script-query variables from runtimeConfig

### ISSUE THREAD

**Issue by JohannesLamberts** [nuxt-community/gtm-module#94]
Currently it is not possible to define query variables at runtime - e.g. in case the same build is used in different environments (gtm_preview).

To solve this, `publicRuntimeConfig.gtm.variables` should override `gtm.variables`, if set.

P.S.: I've got it working since the "id" param is not url-escaped 🤫, but that seems kind of hacky 😉

**Comment by alessandroniciforo** [comment]
+1

**Comment by Martin7mind** [comment]
Yes lets fix this asap! i feel like this is the primary use case for the runtime config?

**Comment by Martin7mind** [comment]
@JohannesLamberts how exactly did you bake it into the ID field?

**Comment by kissu** [comment]
This one is not solving the issue ? If you define it there directly >> https://github.com/nuxt-community/gtm-module/issues/69

**Comment by Martin7mind** [comment]
@kissu that feature does not work properly

**Comment by sergiobellon** [comment]
this is an issue for us too, let's fix it asap please 🙏

**Comment by Martin7mind** [comment]
@sergiobellon this is a workaround:

```
publicRuntimeConfig: {
    gtm: {
      // hack until https://github.com/nuxt-community/gtm-module/issues/94 is fixed
      id: `GTM-xxxx&gtm_auth=${process.env.GTM_AUTH}&gtm_preview=${process.env.GTM_PREVIEW}&gtm_cookies_win=x`
    }
  }
```

**Comment by kissu** [comment]
@Martin7mind this is working with runtime variables, right ?

**Comment by Martin7mind** [comment]
correct @kissu

**Comment by matthew-inamdar** [comment]
+1 This is a must for config driven single builds. @Martin7mind solution works in the meantime.
